### PR TITLE
Fixing Error throwing for forgot+reset password component

### PR DESCRIPTION
**Pull request by jmulholland4014**
On top of the error in the story, I also fixed an error where forgot passwords error was not specific enough, and where the reset passwords error would get overridden by "An error occurred". 
Story: https://app.clubhouse.io/clarkcan/story/4755/password-does-not-meet-requirements-error-when-successfully-resetting-password

### CODE REVIEW

**Review by pzalep1 — APPROVED** [review]

**Review by Cwagne17 — APPROVED** [review]

File: src/app/auth/reset-password/reset-password.component.ts [hunk 1]
```diff
@@ -35,6 +34,7 @@ export class ResetPasswordComponent implements OnInit {
 
     if (!goodPassword) {
       this.error('Password is not strong enough. Must contain 1 lowercase, 1 uppercase, 1 number, and 1 symbol');
```
Comment (mfranz1) [inline src/app/auth/reset-password/reset-password.component.ts:36]:
```suggestion
      return this.error('Password is not strong enough. Must contain 1 lowercase, 1 uppercase, 1 number, and 1 symbol');
```

Comment (mfranz1) [reply src/app/auth/reset-password/reset-password.component.ts:36]: The remove the return on line 37

File: src/app/auth/forgot-password/forgot-password.component.ts [hunk 2]
```diff
@@ -25,14 +25,12 @@ export class ForgotPasswordComponent implements OnInit {
     this.auth.initiateResetPassword(this.email).subscribe(val => {
       this.done = true;
     }, error => {
-      console.log(error);
-      this.error(error.error);
+      this.error(error);
```
Comment (mfranz1) [inline src/app/auth/forgot-password/forgot-password.component.ts:28]: Is error a string or an object?

Comment (jmulholland4014) [reply src/app/auth/forgot-password/forgot-password.component.ts:28]: In this case it's a string. Unlike in CARD the error message is stored in error.

File: src/app/auth/reset-password/reset-password.component.ts (hunk 1 above)
Comment (mfranz1) [reply src/app/auth/reset-password/reset-password.component.ts:36]: I'm convinced it's fine

**Review by mfranz1 — APPROVED** [review]

**Review by EVogel1999 — APPROVED** [review]

**Pull request merged**